br/pkg/backup/client.go

```diff
@@ -533,14 +542,70 @@ func BuildBackupRangeAndSchema(
 			continue
 		}
 
-		tables, err := m.ListTables(dbInfo.ID)
+		tableNum := 0
+		err = m.IterTables(dbInfo.ID, func(tableInfo *model.TableInfo) error {
+			if !tableFilter.MatchTable(dbInfo.Name.O, tableInfo.Name.O) {
+				// Skip tables other than the given table.
+				return nil
+			}
+
+			schemasNum += 1
```

Comment:
Why schemaNum increasing at the same frequency as tableNum?

Reply:
`schemaNum` is to record the total number of `(dbInfo, tableInfo)` or `(dbInfo, nil)`.
`tableNum` is to record whether the `dbInfo` has records any `tableInfo`s.